core/src/main/java/org/mule/TransformationService.java

```diff
@@ -0,0 +1,513 @@
+import java.util.List;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ * Provides the same operations previously exposed by {@link MuleMessage} but decoupled from MuleMessage.
+ *
+ * TODO Redefine this interface as part of Mule 4.0 transformation improvements.
+ */
+public class TransformationService
+{
+
+    private static final Log logger = LogFactory.getLog(TransformationService.class);
+
+    private MuleContext muleContext;
+
+    public TransformationService(MuleContext muleContext)
+    {
+        this.muleContext = muleContext;
+    }
+
+    /**
+     * Applies a list of transformers to the payload of the message. This *Will* change the payload of the
+     * message.
+     *
+     * @param event the event being processed
+     * @param transformers the transformers to apply to the message payload
+     * @throws TransformerException if a transformation error occurs or one or more of the transformers passed in a
+     * are incompatible with the message payload
+     */
+    public void applyTransformers(MuleMessage message, MuleEvent event, List<? extends Transformer> transformers) throws MuleException
+    {
+        applyTransformers(message, event, transformers, null);
+    }
+
+    /**
+     * Applies a list of transformers to the payload of the message. This *Will* change the payload of the
+     * message.
+     *
+     * @param event the event being processed
+     * @param transformers the transformers to apply to the message payload
+     * @throws TransformerException if a transformation error occurs or one or more of the transformers passed in a
+     * are incompatible with the message payload
+     */
+    public void applyTransformers(MuleMessage message, MuleEvent event, Transformer... transformers) throws MuleException
+    {
+        applyTransformers(message, event, Arrays.asList(transformers), null);
+    }
+
+    /**
+     * Applies a list of transformers to the payload of the message. This *Will* change the payload of the
+     * message.
+     *
+     * @param event the event being processed
+     * @param transformers the transformers to apply to the message payload
+     * @param outputType the required output type for this transformation. by adding this parameter some additional
+     * transformations will occur on the message payload to ensure that the final payload is of the specified type.
+     * If no transformers can be found in the registry that can transform from the return type of the transformation
+     * list to the outputType and exception will be thrown
+     * @throws TransformerException if a transformation error occurs or one or more of the transformers passed in a
+     * are incompatible with the message payload
+     */
+    public void applyTransformers(MuleMessage message, MuleEvent event, List<? extends Transformer> transformers, Class<?> outputType) throws MuleException
+    {
+        if (!transformers.isEmpty())
```

Comment:
Ideally, but i'd prefer not to change transformation behavior as part of this refactor.  This will be looked at as part of MULE-9141 surely.